File: typescript/src/flight.ts
export class Flight {
  constructor(
    public readonly airline: string,
    public readonly flightNumber: string,
    public readonly price: number,
    public readonly departureTime: string,
    public readonly arrivalTime: string,
    public readonly duration: string,
  ) {}

  toString(): string {
    return `Flight{airline='${this.airline}', flightNumber='${this.flightNumber}', price=${this.price}, departureTime='${this.departureTime}', arrivalTime='${this.arrivalTime}', duration='${this.duration}'}`;
  }
}

File: rust/src/flight.rs
/// Represents a single flight option.
#[derive(Debug, Clone, PartialEq)]
pub struct Flight {
    pub airline: String,
    pub flight_number: String,
    pub price: u32,
    pub departure_time: String,
    pub arrival_time: String,
    pub duration: String,
}

impl std::fmt::Display for Flight {
    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
        write!(
            f,
            "Flight {{ airline: {:?}, flight_number: {:?}, price: {}, \
             departure_time: {:?}, arrival_time: {:?}, duration: {:?} }}",
            self.airline,
            self.flight_number,
            self.price,
            self.departure_time,
            self.arrival_time,
            self.duration,
        )
    }
}
